lib/internal/repl.js

```diff
@@ -127,7 +127,8 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     }
 
     if (data) {
-      repl.history = data.split(/[\n\r]+/, repl.historySize);
+      repl.history = data.trim().split(/[\n\r]+/)
+        .reverse().slice(0, repl.historySize);
```

Comment:
@addaleax @Fishrock123 historically, `historySize` has limited both the file size, and the maximum number of lines loaded into memory at start up. But this almost seems like a side effect of `readline` usage, and is not documented anywhere that I could find in the context of the REPL.

This PR changes it so that so that the number of lines stored in memory from the history file at start up is still limited, but the history file size itself is unbounded.